pkg/syncer/shared/helpers.go

```diff
@@ -54,3 +60,14 @@ func GetUpstreamResourceName(downstreamResourceGVR schema.GroupVersionResource,
 	}
 	return downstreamResourceName
 }
+
+// GetDNSID returns a unique ID for DNS object derived from the sync target name, its UID and workspace. It's
+// a valid DNS segment and can be used as namespace or object names.
+func GetDNSID(workspace logicalcluster.Name, syncTargetUID types.UID, syncTargetName string) string {
+	syncerHash := sha256.Sum224([]byte(syncTargetUID))
+	uid36hash := strings.ToLower(base36.EncodeBytes(syncerHash[:]))
+	workspaceHash := sha256.Sum224([]byte(workspace.String()))
+	workspace36hash := strings.ToLower(base36.EncodeBytes(workspaceHash[:]))
+
+	return fmt.Sprintf("kcp-dns-%s-%s-%s", syncTargetName, uid36hash[:8], workspace36hash[:8])
```

Comment:
Do we ensure that the result will never exceed the size limit for names ?

Reply:
Here, the only unconstraint size is `syncTargetName` and I don't know if its size is limited. I'll have to check.

Reply:
found it: https://github.com/kcp-dev/kcp/blob/main/pkg/cliplugins/workload/plugin/sync.go#L177

fixing...